### ISSUE TITLE
Deploying a model with kfserving continuously deploys 100's of models

### ISSUE THREAD

**Issue by rileygreen022** [kserve/kserve#1864]
When I deploy a custom container mode on kfserving, it continuously deploys 100's of models until the model is deleted. 

```
namespace = 'modelserve'
service_name = "kfserving-custom-model"

predictor = V1beta1PredictorSpec(
    min_replicas=1,
    containers=[{
        "image":"docker.io/XXX/XXX:latest",
         "name":service_name,
        "resources": {
            "limits": {"cpu": "500m", "memory": "4Gi"},
            "requests": {"cpu": "250m", "memory": "2Gi"},
        }
    }],
)

isvc = V1beta1InferenceService(
    api_version=constants.KFSERVING_V1BETA1,
    kind=constants.KFSERVING_KIND,
    metadata=client.V1ObjectMeta(
        name=service_name, namespace=namespace
    ),
    spec=V1beta1InferenceServiceSpec(predictor=predictor),
)
```

<img width="1392" alt="Screen Shot 2021-10-09 at 9 12 56 pm" src="https://user-images.githubusercontent.com/24871158/136654248-e1e7c5df-858a-47d1-80c1-2a7a3e503069.png">

<img width="1039" alt="Screen Shot 2021-10-09 at 9 12 18 pm" src="https://user-images.githubusercontent.com/24871158/136654250-f64cc712-6721-4745-98c7-32ffea7aadff.png">

![Screen Shot 2021-10-09 at 9 23 20 pm](https://user-images.githubusercontent.com/24871158/136654305-bf0687c9-9ad0-455b-a043-e10c52d05a9d.png)

Envrionment: Azure AKS
Kubeflow Config: https://raw.githubusercontent.com/kubeflow/manifests/v1.2-branch/kfdef/kfctl_istio_dex.v1.2.0.yaml 
KFserving Config: https://github.com/kubeflow/kfserving/releases/download/v0.6.0/kfserving.yaml 

Notes:
- The test docker container is public, so access should not be an issue.
- I tested this exact set-up a few months ago and it all worked perfectly.
- I have reinstalled the environment 5+ times and have not been able to get it to work. 
- I have tried setting the max number of replicas to 2, this does not stop it. 
- I run the above code from a Jupiter notebook on kubeflow. 

Let me know if you require any particular logs and the command to get the logs please :)

**Comment by yuzisun** [comment]
@rileygreen022 Did you install KServe with kubeflow or standalone, seems like you installed two version of KServe?

**Comment by rileygreen022** [comment]
Ah I see, when I tried to deploy a model with the kfctl_istio_dex.v1.2.0.yaml standalone, I don't think it had the V1beta1 API.
Should I remove the following lines below when installing and then complete the same steps installing 0.0.6?

```
  - kustomizeConfig:
      repoRef:
        name: manifests
        path: kfserving/installs/generic
    name: kfserving
```

Anything else I need to do?

**Comment by yuzisun** [comment]
@rileygreen022 I'd highly recommend you use kubeflow 1.3 or recently released 1.4 which gets you to v1beta1 API
